Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListView.swift

```diff
@@ -45,72 +45,86 @@ public struct ChatChannelListView<Factory: ViewFactory>: View {
-                        viewModel: viewModel,
-                        onItemTap: onItemTap
-                    )
-                }
-            }
-            .onDisappear(perform: {
-                if viewModel.selectedChannel != nil {
-                    viewModel.hideTabBar = true
-                }
-                if viewModel.swipedChannelId != nil {
-                    viewModel.swipedChannelId = nil
-                }
+        container()
+            .overlay(viewModel.customAlertShown ? customViewOverlay() : nil)
+            .accentColor(colors.tintColor)
+            .if(isIphone || !utils.messageListConfig.iPadSplitViewEnabled, transform: { view in
+                view.navigationViewStyle(.stack)
             })
             .background(
-                viewFactory.makeChannelListBackground(colors: colors)
-            )
-            .alert(isPresented: $viewModel.alertShown) {
-                switch viewModel.channelAlertType {
-                case let .deleteChannel(channel):
-                    return Alert(
-                        title: Text(L10n.Alert.Actions.deleteChannelTitle),
-                        message: Text(L10n.Alert.Actions.deleteChannelMessage),
-                        primaryButton: .destructive(Text(L10n.Alert.Actions.delete)) {
-                            viewModel.delete(channel: channel)
-                        },
-                        secondaryButton: .cancel()
-                    )
-                default:
-                    return Alert.defaultErrorAlert
-                }
-            }
-            .modifier(viewFactory.makeChannelListHeaderViewModifier(title: title))
-            .navigationBarTitleDisplayMode(viewFactory.navigationBarDisplayMode())
-            .blur(radius: (viewModel.customAlertShown || viewModel.alertShown) ? 6 : 0)
-        }
-        .overlay(viewModel.customAlertShown ? customViewOverlay() : nil)
-        .accentColor(colors.tintColor)
-        .if(isIphone || !utils.messageListConfig.iPadSplitViewEnabled, transform: { view in
-            view.navigationViewStyle(.stack)
-        })
-        .background(
-            isIphone ?
+                isIphone ?
                 Color.clear.background(
                     TabBarAccessor { tabBar in
                         self.tabBar = tabBar
                     }
                 )
                 .allowsHitTesting(false)
                 : nil
+            )
+            .onReceive(viewModel.$hideTabBar) { newValue in
+                if isIphone {
+                    self.setupTabBarAppeareance()
+                    self.tabBar?.isHidden = newValue
+                }
+            }
+            .accessibilityIdentifier("ChatChannelListView")
+    }
+    
+    @ViewBuilder
+    private func container() -> some View {
+        if utils.messageListConfig.embedInNavigationView == true {
+            NavigationView {
+                content()
+            }
+        } else {
+            content()
```

Comment:
Would be nice to have a snapshot test for this case.

Reply:
Added a snapshot test case. Not 100% sure if I did this right, but the snapshot looks correct and the test case passes.